src/services/gemini.py

```diff
@@ -0,0 +1,73 @@
+
+import google.generativeai as genai
+import asyncio, os
+import logging
+import time
+from ratelimit import limits, sleep_and_retry
+
+# Setup logging
+logging.basicConfig(level=logging.ERROR, format="%(asctime)s, %(levelname)s: %(message)s")
+
+class GeminiPro:
+    def __init__(self):
+        # Load API key
+        api_key = os.environ.get("GEMINI_API_KEY")
```

Comment:
Here we are getting "GEMINI_API_KEY" directly from os.environ, which is okay, but for maintainability, keys and secret should be instead imported from ../utlis/config.py. If you are unsure how to import keys from config.py, you can check queryLangchain.py as reference.

Reply:
This has been fixed in the new commit. GEMINI_API_KEY is now imported using utils.config.